docs/master/federation/reference-resolvers.md

```diff
@@ -0,0 +1,211 @@
+     */
+    public function __invoke(array $representations): iterable
+    {
+        // TODO return multiple values that match type Foo
+    }
+}
+```
+
+The returned iterable _must_ have the same keys as the given `array $representations` to enable Lighthouse
+to return the results in the correct order.
+
+```php
+namespace App\GraphQL\ReferenceResolvers;
+
+use App\Repositories\SomeProductRepository;
+use Illuminate\Support\Arr;
+use Nuwave\Lighthouse\Federation\BatchedEntityResolver;
+
+final class Product implements BatchedEntityResolver
+{
+    public function __invoke($representation): iterable
+    {
+        $result = [];
+        $ids = Arr::pluck($representation, 'id');
+        $products = SomeProductRepository::getSomeProductsCollection($ids);
+
+        foreach ($representation as $hash => $representation) {
+            $id = Arr::get($representation, 'id');
+
+            $result[$hash] = $products->where('id', $id)->first(); // returned iterable _must_ have the same keys as the given `array $representations`
+        }
+
+        return $result;
+    }
+}
+```
+
+### Base Batched Entity Resolvers
+
+For convenience, you can use a base class BaseBatchedReferenceResolver. He uses the `__invoke()` method to prepare the
+input request and generate the response, as well as provides some auxiliary methods. So all you need to do is inherit
+from the base class and implement the `resolve()` method, which should return a collection of results. The base class
+will be responsible for generating the response.
+
+```php
+namespace App\GraphQL\ReferenceResolvers;
+
+use App\Models\Product;
+use Illuminate\Support\Arr;
+use Nuwave\Lighthouse\Federation\Resolvers\BaseBatchedReferenceResolver;
+
+final class Product extends BaseBatchedReferenceResolver
+{
+    public function resolve(): Collection
+    {
+        $primaryKey = $this->getPrimaryKey();
+        $ids = $this->getRepresentationIds()
+    
+        return Product::whereIn($primaryKey, $ids)->get();
+    }
+}
+```
+
+## Eloquent Model Resolvers
+
+When no resolver class can be found, Lighthouse will attempt to find the model that
+matches the type `__typename`, using the namespaces configured in `lighthouse.namespaces.models`.
+
+```graphql
+{
+    _entities(representations: [{ __typename: "Foo", bar: "asdf", baz: 42 }]) {
+        ... on Foo {
+            id
+        }
+    }
+}
+```
+
+The additional fields in the representation constrain the query builder, which is then
+called and expected to return a single result.
+
+```php
+$results = App\Models\Foo::query()
+    ->where('bar', 'asdf')
+    ->where('baz', 42)
+    ->get();
+
+if ($results->count() > 1) {
+    throw new Error('The query returned more than one result.');
+}
+
+return $results->first();
+```
+
+For model resolvers, it is typical that they make one query to the database for each entity, which can lead to the "n+1"
+problem. Therefore, for a large number of entities, it is worth
```

Comment:
```suggestion
The default model resolver makes one database query for each entity.
Therefore, for a large number of entities, it is worth
```

There is no question about it, they definitely do. Also, this is not a case of n+1, it is just n.